### PR TITLE
docs(cwpool): better document pool models

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #6316

## What is the purpose of the change

Better docs for cw pool models

### CODE REVIEW

File: proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto [hunk 1]
```diff
@@ -8,6 +8,21 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/osmosis-labs/osmosis/v19/x/cosmwasmpool/model";
 
+// CosmWasmPool represents the data serialized into state for each CW pool.
+//
+// Note: CW Pool has 2 pool models:
+// - CosmWasmPool which is a proto-generates store model used for serialization
```
Comment (mattverse) [inline proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto:14]:
```suggestion
// - CosmWasmPool which is a proto-generated store model used for serialization
```

File: proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto [hunk 2]
```diff
@@ -8,6 +8,21 @@ import "google/protobuf/timestamp.proto";
...
+// - Pool struct that encapsulates the CosmWasmPool and wasmKeeper for calling
+// the contract.
+//
+// CosmWasmPool implements the poolmanager.PoolI interface but it panics on all
+// methods. The reason is that access to wasmKeeper is required to call the
+// contract.
+//
+// Instead, all interactions and poolmanager.PoolI methods are to be performed
+// on the Pool struct. The reason why we cannot have a Pool struct only is
+// because it cannot be serialized into state.
```
Comment (mattverse) [inline proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto:25]: Just out of curiosity, why not?

**Review by mattverse — APPROVED** [review]

File: proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto (hunk 1 above)
Comment (p0mvn) [reply proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto]: Fixed: [f42460a](https://github.com/osmosis-labs/osmosis/pull/6484/commits/f42460a65181e7929b992b729a756930eb718744)

File: proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto (hunk 2 above)
Comment (p0mvn) [reply proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto:25]: I will elaborate in the comment.

The reason is that it has a wasmKeeper as field that is non-serializable.

Comment (p0mvn) [reply proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto]: [f72cbc6](https://github.com/osmosis-labs/osmosis/pull/6484/commits/f72cbc6710dd2e4d6ebba3cd1ec29f32ab173139)

**Pull request merged**